addressed all warnings

grendell opened the pull request:
* Addressed warnings for lost const qualifiers, unreferenced arguments, implicit declarations, and various overflows.
* Silenced warnings for unreachable code and conditional flow changed by optimizer.
  * sdcc 4.5.0 does not offer support for disabling warnings in a portion of a file, so these are disabled for all of encounter.c and main.c with this change.
* Attempted to add `--Werror` to prevent warnings from creeping back in, but this is not honored by sdcc 4.5.0, at least on macos-arm64.
* Tested by completing floor 1 and pulling levers in floor 2.

rsandor commented on src/encounter.c:
```diff
@@ -1,5 +1,8 @@
 #pragma bank 3
 
+// disable "conditional flow changed by optimizer" warning due to jump table optimization in update_player_status_effects()
+#pragma disable_warning 110
```
I don't think we should disable this one, because I've definitely caught many bugs during development because of it. I think it is okay to let it slide or we should rework the `update_playe_status_effects()`.

rsandor commented on src/floor3.c:
```diff
@@ -179,6 +180,7 @@ static const Door doors[] = {
 //------------------------------------------------------------------------------
 
 static void on_lit(const Sconce* sconce) {
```
We should just get rid of this entirely since it's not used on this floor.

rsandor commented on src/floor6.c:
```diff
@@ -234,6 +236,7 @@ static const Door doors[] = {
 //------------------------------------------------------------------------------
 
 static void on_lit(const Sconce* sconce) {
```
Similar, just remove the function entirely.

rsandor commented on src/main.c:
```diff
@@ -1,3 +1,9 @@
+// disable "unreachable code" warning since initial_mode is const.
+#pragma disable_warning 126
```
Let's remove these disables, I was planning on reworking this section anyway (#74).

rsandor commented on src/map.c:
```diff
@@ -2123,7 +2123,7 @@ static void redraw_tile(uint8_t map_id, int8_t x, int8_t y) {
   if (x < 0 || y < 0)
     return;
-  if (x >= active_map->width || y >= active_map->height)
+  if (x >= (int8_t)active_map->width || y >= (int8_t)active_map->height)
```
This could cause a weird bug if somehow `active_map->width > 127`. It probably doesn't matter but might want to add a comment in case we rework the maps to allow for much larger sizes.

grendell replied:
I think it might be best to explicitly cast both to int in that case, which is what the compiler was implicitly doing originally.  It'll be a slightly more expensive check, but safer.

Out of curiosity, when can x or y be negative?  If those could be unsigned that would also be a simpler fix.

rsandor replied:
X and Y can be negative if the player is in the upper left portion of the screen. I chose to let this happen because it made the logic for handling the rendering a bit easier (I look for < 0 and just render it as "BLANK" iirc).

Either way I think your fix is right, just wanna make sure we document that this is happening somewhere less opaque than an if statement buried in `map.c`.

grendell replied:
Ah, that makes sense, and is a cool way to handle off-the-map tiles!  Thanks for the context.

grendell commented:
* Addressed all feedback.
* Removed `--Werror`.
  * Filed a [bug](https://sourceforge.net/p/sdcc/bugs/3868/) against sdcc with regards to pragma save/pragma restore not affecting pragma disable_warning.

rsandor commented:
Looks great! Merging 😄
